app/controllers/api/v0/service_plans_controller.rb

```diff
@@ -19,19 +21,40 @@ def order
 
       private
 
-      def order_service_plan_and_update_task(task, service_plan)
-        task.context = service_plan.order(order_params.slice(:service_parameters, :provider_control_parameters))
-        task.status = "completed"
-        task.save!
+      def service_plan_id
+        params[:service_plan_id].to_i
       end
 
-      def order_params
+      def params_for_order
         params.permit(
           :service_plan_id,
           :service_parameters          => {},
           :provider_control_parameters => {}
         ).to_h
       end
+
+      def params_for_list
+        params.permit(:source_id, :tenant_id, :service_offering_id)
+      end
+
+      def messaging_client
+        require "manageiq-messaging"
+
+        ManageIQ::Messaging::Client.open(messaging_opts)
+      end
+
+      def messaging_opts
+        @messaging_opts ||= {
+          :protocol => :Kafka,
+          :host     => ENV["QUEUE_HOST"] || "localhost",
+          :port     => ENV["QUEUE_PORT"] || "9092",
+          :encoding => "json"
+        }
+      end
+
+      def model
```

Comment:
Why is this being added here? Wasn't there a call to `.model` before this PR?

Reply:
Yeah, this is all defined here https://github.com/ManageIQ/topological_inventory-api/blob/master/app/controllers/application_controller.rb#L21-L23

It works as is:
```
irb(main):005:0> Api::V0::ServicePlansController.send(:model)
=> ServicePlan (call 'ServicePlan.connection' to establish a connection)
```